Modules/Projects/Core/Script_MiscUtilities.lua

```diff
@@ -134,6 +134,19 @@ function Rest()
     Player.Health = Player.HealthMax
 end
 
+function RestUntilNight()
+    Sleep(1)
+
+    if GetIsNight() then
+        Log("SLEEP_NIGHT_CANNOT")
+    else
+        SetTimeHour(20)
```

Comment:
This should sleep until 21. `GetIsNight()` will return false at hour 20.

Reply:
corrected and pushed again.